models/azure_project.rb

```diff
@@ -548,16 +550,19 @@ def api_query_compute_nodes
       if response.success?
         vms = response['value']
         vms.select { |vm| vm.key?('tags') && vm['tags']['type'] == 'compute' && self.resource_groups.include?(vm['id'].split('/')[4].downcase) }
-      elsif response.code == 504 && attempt < MAX_API_ATTEMPTS
+      elsif response.code == 504
         raise Net::ReadTimeout
       else
-        raise AzureApiError.new("Error querying compute nodes for project #{name}.\nError code #{response.code}.\n#{response if @verbose}")
+        raise AzureApiError.new("Error querying compute nodes for project #{name}."\
+                                "\nError code #{response.code}.\n#{response if @verbose}")
       end
     rescue Net::ReadTimeout
       if attempt < MAX_API_ATTEMPTS
+        error.error_messages.append("Attempt #{attempt}:\nError code #{response.code}."\
```

Comment:
Sorry, another thing I've just noticed - if this is reached by a `Net::ReadTimeout` error that we haven't raised ourselves, then `response` may be nil, so this will break. So the error code bit needs to be added only if `response` is present.